### PR TITLE
feat: update dynamic form to handle generic form template

### PR DESCRIPTION
**Pull request by isaackps**
### in this PR:
- update the generic form template in the config file
- added custom text area widget so that we can have text area in the generic form template.

Screenshot 2020-09-09 at 5 27 30 PM

### CODE REVIEW

File: src/components/DynamicFormContainer/DynamicForm/CustomTemplates/CustomTextareaWidget.test.tsx [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    rawErrors: [],
+  };
+};
+
+describe("customTextareaWidget", () => {
+  it("should render the correct textarea", () => {
+    render(<CustomTextareaWidget {...propsToPassIntoWidget()} />);
+
+    expect(screen.queryAllByText("some default text")).toHaveLength(1);
+    expect(screen.queryAllByText("some placeholder text")).toHaveLength(0);
```
Comment (simboonlong) [inline src/components/DynamicFormContainer/DynamicForm/CustomTemplates/CustomTextareaWidget.test.tsx:43]: this is testing nothing i think, it should be:

```suggestion
expect(screen.queryAllByPlaceholderText("some placeholder text")).toHaveLength(1);
```

yeah?

File: src/components/DynamicFormContainer/DynamicForm/CustomTemplates/CustomTextareaWidget.tsx [hunk 2]
```diff
@@ -0,0 +1,37 @@
+import React, { ChangeEvent, FunctionComponent } from "react";
+import { WidgetProps } from "react-jsonschema-form";
+
```
Comment (simboonlong) [inline src/components/DynamicFormContainer/DynamicForm/CustomTemplates/CustomTextareaWidget.tsx:3]: felt that this code writing style was out of place, think u may want to put a comment linking to the source code which u referred to?

File: src/components/DynamicFormContainer/DynamicForm/CustomTemplates/CustomTextareaWidget.tsx [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+  disabled,
+  readonly,
+  autofocus,
+  onChange,
+  onBlur,
+  onFocus,
+}) => {
+  const _onChange: (event: ChangeEvent<HTMLTextAreaElement>) => void = ({ target: { value } }) => {
+    return onChange(value === "" ? options.emptyValue : value);
+  };
```
Comment (simboonlong) [inline src/components/DynamicFormContainer/DynamicForm/CustomTemplates/CustomTextareaWidget.tsx:19]: felt that this particular file's code writing style was out of place, think u may want to put a comment at the very top, linking to the source code which u referred to?

File: src/components/DynamicFormContainer/DynamicForm/CustomTemplates/CustomTextareaWidget.test.tsx (hunk 1 above)
Comment (isaackps) [reply src/components/DynamicFormContainer/DynamicForm/CustomTemplates/CustomTextareaWidget.test.tsx:43]: ok will update

File: src/components/DynamicFormContainer/DynamicForm/CustomTemplates/CustomTextareaWidget.tsx (hunk 3 above)
Comment (isaackps) [reply src/components/DynamicFormContainer/DynamicForm/CustomTemplates/CustomTextareaWidget.tsx:19]: ok will add it in

**Review by simboonlong — APPROVED** [review]

**Comment by simboonlong** [comment]
looks okay

**Pull request merged**